src/Components/FirmwareSelector/index.jsx

```diff
@@ -374,12 +370,11 @@ FirmwareSelector.propTypes = {
     versions: PropTypes.shape().isRequired,
     pwm: PropTypes.shape().isRequired,
   }).isRequired,
-  escHint: PropTypes.string,
+  esc: PropTypes.shape(),
```

Comment:
Please also add the properties that are used from the esc shape to the validation. Default prop for ESC could then be an object with the matching properties and we could skip the property checks in the component.